### ISSUE TITLE
Internal server error when sending duplicate cookie

### ISSUE THREAD

**Issue by viniciusffj** [tomakehurst/wiremock#567]
Hi.

I'm experiencing a strange error in wiremock.

Let's say I have the following mapping:
```json
{
    "request": {
        "method": "GET",
        "url": "/test"
    },
    "response": {
        "status": 200,
        "body": "More content\n"
    }
}
```

If I make a request with two values for one cookie, like this:
```bash
$ curl 'http://localhost:8080/test' -H 'Cookie: cookie1=190283aJDS;cookie1=190283aJDS'
```

I get the following response:
```html
<html>

<head>
    <meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1" />
    <title>Error 500 </title>
</head>

<body>
    <h2>HTTP ERROR: 500</h2>
    <p>Problem accessing /test. Reason:
        <pre>    java.lang.IllegalArgumentException: Multiple entries with same key: cookie1=190283aJDS and cookie1=190283aJDS</pre></p>
    <hr /><i><small>Powered by Jetty://</small></i>
</body>

</html>
```

But if I remove the duplication, it works. For example:
```bash
$ curl 'http://localhost:8080/test' -H 'Cookie: cookie1=190283aJDS'
```

I know that we shouldn't set the cookie value twice, but in my real life case I really can't control that.
What could I do?

Also, I've tried to debug wiremock but I didn't go too far. Is there any debug flag (I'm already using --verbose) that I can see the stacktrace?

**Comment by tomakehurst** [comment]
Yeah, poor modelling decision on my part. Cookies are represented by a map internally and the map builder blows up if you try to add more than one with the same key.

I'll switch to a list when I get the chance.

**Comment by viniciusffj** [comment]
Thanks.

If you need help I can try to make a PR, but I need some guidance first given that I'm not familiar with the code.

**Comment by tomakehurst** [comment]
If you fancy having a go at it, it's primarily about changing `Request.getCookies()` to return a `List<Cookie>` rather than a map, changing `WireMockHttpServletRequestAdapter` so it adapts to a list (and therefore duplicates being OK), and dealing with the consequences of the switch in the handful of features that read cookies.

**Comment by viniciusffj** [comment]
Ok, I'll take a look on those and try to open a PR.

**Comment by mmodenesi** [comment]
Hi, I'm facing the same problem. Can you suggest any workarround for this issue without changing the version of wiremock? Thanks in advance!

**Comment by viniciusffj** [comment]
Hi @mmodenesi 

I solved the problem by removing the duplicated cookies in my front end app.

Unfortunately I failed to submit a PR to fix this, I'll try again when I have a free time

**Comment by mmodenesi** [comment]
Thank you. Going that way I found that the issue was related to Firefox 46.0 sending duplicate cookies, not my application. Upgrading the browser solved the issue.

**Comment by tomakehurst** [comment]
Fixed in 827e7ccb3edcb15bb78e68add74adee536e921b2
